invenio_circulation/transitions/transitions.py

```diff
@@ -317,14 +296,25 @@ class ItemInTransitHouseToItemReturned(Transition):
     @ensure_same_item
     def before(self, loan, **kwargs):
         """Validate check-in action."""
-        super(ItemInTransitHouseToItemReturned, self).before(loan, **kwargs)
+        super().before(loan, **kwargs)
+        self.assign_item = kwargs.get('assign_item', True)
 
         _ensure_same_location(loan['item_pid'],
                               loan['transaction_location_pid'],
                               self.dest,
                               error_msg="Item should be in transit to house. ")
 
     def after(self, loan):
-        """Convert dates to string before saving loan."""
-        super(ItemInTransitHouseToItemReturned, self).after(loan)
-        _update_document_pending_request_for_item(loan['item_pid'])
+        """Check for pending requests on this item after check-in."""
+        super().after(loan)
+        if self.assign_item:
+            _update_document_pending_request_for_item(loan['item_pid'])
+
+
+class ToCancelled(Transition):
```

Comment:
without this, user could by mistake cancel the loan and override the item of the loan